### ISSUE TITLE
Fix Request: false positives with LimitStatementsInLambda

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#532]
The rule produces more violations than in the pmd6 version, because the number of Statements is counted differently.  

Example:
```
class Foo {
    public Flux<Result> getBalances(final BalanceList request) throws BalanceException {
        return Flux.deferContextual(ctx -> {
            try { // stmt1 // <- false positive
                return webClient.post() // stmt2
                        .uri(ovConfigProperties.getUrl())
                        .body(BodyInserters.fromValue(request))
                        .headers(h -> h.addAll(ovHttpHeaders))
                        .retrieve()
                        .bodyToFlux(Result.class);
            } catch(BalanceException e){ 
                log.error("Exception in OV Rest call {}", e); // stmt3
                return Flux.empty(); // stmt4
            }
        });
    }
}
```
The rule checks for > 5 statement and produces a violation, while I count only 4 statements in the above example.

**Comment by jborgers** [comment]
Turns out Block is defined as Statement, we should not count those.
https://docs.pmd-code.org/apidocs/pmd-java/7.10.0/net/sourceforge/pmd/lang/java/ast/ASTStatement.html
